plugin/cmd.go

```diff
@@ -18,40 +18,67 @@ const (
 	PluginMagicCookieKey       = "INGEST_PLUGIN"
 )
 
-func RunPluginServer(s Source, d Destination, g prometheus.Gatherer) {
+type Option func(c *configuration)
+
+func WithLogger(l hclog.Logger) Option {
+	return func(c *configuration) {
+		c.l = l
+	}
+}
+
+func WithGatherer(g prometheus.Gatherer) Option {
+	return func(c *configuration) {
+		c.g = g
+	}
+}
+
+type configuration struct {
+	g prometheus.Gatherer
+	l hclog.Logger
+}
+
+func RunPluginServer(s Source, d Destination, opts ...Option) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
+	c := &configuration{
```

Comment:
Do we need to default the gatherer too?

Reply:
Like, will the plugin panic if it tries to gather and we didn't pass an optional gatherer during instantiation?

Reply:
Because since WithGatherer is an option i dont think it should be required

Reply:
the plugin server will create a new Registry when the Gatherer is nil. Of course we can move that logic from plugin.go to cmd.go and remove the nil check.

Reply:
it seems like it would be cleaner and would make sense if these defaulting checks were consolidated in one place

Reply:
ok done

Reply:
we could remove some code, but now plugins must register the go and process collector if they use their own registry